WordPress/gutenberg
# Fix/screen reader text

## Description

This PR solves https://github.com/WordPress/gutenberg/issues/18167 

See #18022 which introduced the VisuallyHidden component. The usage of `screen-reader-text` assumes a WordPress context with the proper CSS already defined. This may not always be the case, especially when pieces of the Gutenberg project are used elsewhere.

## How has this been tested?

Tested locally. This PR touches a number of blocks and components. For each of them, you need to check if the new `<VisuallyHidden>` part behaves in the same way as the old `.screen-reader-text`. For example, this PR touches a gallery block caption:

https://github.com/WordPress/gutenberg/pull/20607/files#diff-e25f8ba6320e6c1650e6ec1c0ae5ba34L43-L102

To test it, let's follow some steps:

1. Create a new gallery block
1. Confirm the caption is hidden when it's empty and the block is not selected
1. Confirm the caption is visible then it's empty and the block is selected
1. Confirm the caption is visible when it's not empty and the block is not selected
1. Confirm the caption is visible then it's not empty and the block is selected

Without this PR applied it should use CSS class `screen-reader-text` [(gif)](https://78h87h.files.wordpress.com/2020/03/2020-03-03-15-26-31.2020-03-03-15_59_09.gif)

With this PR applied it should use the new `VisuallyHidden` wrapper [(gif)](https://78h87h.files.wordpress.com/2020/03/2020-03-03-15-35-58.2020-03-03-15_43_45.gif)

## Types of changes
Non-breaking change which fixes #18167 

## Checklist:
